geode-core/src/main/java/org/apache/geode/cache/configuration/CacheConfig.java

```diff
@@ -2612,7 +2612,8 @@ public void setOverflowDirectory(String value) {
   @XmlAccessorType(XmlAccessType.FIELD)
   @XmlType(name = "", propOrder = {"gatewayEventFilters", "gatewayEventSubstitutionFilter",
       "gatewayTransportFilters"})
-  public static class GatewaySender {
+  public static class GatewaySender extends CacheElement {
```

Comment:
You should create a new unit test even for inner-classes. Any class that you make Serializable should also have to unit test coverage to ensure that both seriallization and deserialization succeeds and results in a faithful replica of the original object (shamelessly copied from Effective Java 3rd Ed. Item 86). I recommend using org.apache.commons.lang3.SerializationUtils to simplify the unit testing.

Reply:
Removed serialization impacts, so no need for additional tests.